src/core/dir.c

```diff
@@ -77,8 +86,15 @@ const dir_listing *dir_find_files_with_extension(const char *extension)
         return &listing;
     }
     fs_dir_entry *entry;
+    struct stat file_info;
     while ((entry = fs_dir_read(d)) && listing.num_files < DIR_MAX_FILES) {
         const char *name = dir_entry_name(entry);
+        if (stat(name, &file_info) != -1) {
```

Comment:
We do not need stat for this, the entry contains a d_type variable. Or is that not good enough?

Reply:
MinGW for windows does not have `d_type`, so I had to resort to `stat`